integration-tests/test_collection.py

```diff
@@ -152,8 +159,13 @@ def test_branch_info(insights_client):
     branch_info_path = os.path.join(directory_name, "branch_info")
     with open(branch_info_path, "r") as file:
         data = json.load(file)
-        assert data["remote_branch"] == -1, "Incorrect remote_branch value"
-        assert data["remote_leaf"] == -1, "Incorrect remote_leaf value"
+        if "satellite" in test_config.environment:
+            assert data["product"]["type"] == "Satellite"
+            assert isinstance(uuid.UUID(data["remote_branch"]), uuid.UUID)
+            assert uuid.UUID(data["remote_leaf"]) == subman.uuid
```

Comment:
- the first assertion is a tautology (always true): if the UUID string can be parsed from `data["remote_branch"]`, then the result is always a `uuid.UUID` object
- if `uuid.UUID()` cannot parse a UUID string, it raises an exception (good, will fail the test as expected)
- the second assertion builds again a `uuid.UUID` object

So I'd simplify these two lines a bit:
```python
remote_branch_uuid = uuid.UUID(data["remote_branch"])
assert remote_branch_uuid == subman.uuid
```
- a failing UUID string properly causes `uuid.UUID()` to raise an exception
- only one `uuid.UUID` object constructed
- compare against the sub-man UUID as already done

Reply:
data["remote_branch"] and data["remote_leaf"] represent two different things. 
A branch is an organization for Satellite 6, is a UUID id of the upstream consumer manifest for organization to which host is subscribed. A leaf means subscription manager system identity UUID.

Here I use "assert isinstance(uuid.UUID(data["remote_branch"]), uuid.UUID)" to check the branch info is a uuid object, but I don’t check the exact value of it as the manifest can be re-imported/changed if necessary.
So I just check the leaf value, use "assert uuid.UUID(data["remote_leaf"]) == subman.uuid"

Reply:
pls refer https://docs.google.com/document/d/193mN5aBwxtzpP4U-vnL3yVawPxiG20n0zkzDYsam-eU/edit